app/models/account.rb

```diff
@@ -13,6 +13,18 @@ class Account < ApplicationRecord
 
   validates :account_number, presence: true
 
+  class << self
+    def from_identity_header(identity_header)
+      account = Account.find_or_create_by(
+        account_number: identity_header.identity['account_number']
```

Comment:
Nit: We really ought to extend our identity header validation to include checking for account_number. It could be nil here.

Reply:
I don't really understand :confused: do you mean [this case](https://github.com/RedHatInsights/compliance-backend/pull/984/files#diff-11ba21e2704420a5b829d0b858daa1c977614447eb9ef38f9d792e04e677e62dL39)?

Reply:
I mean to validate account_number within this validation method: https://github.com/RedHatInsights/compliance-backend/blob/master/app/models/identity_header.rb#L18

Not a part of this PR, just pointing out that there isn't any guarantee `account_number` is non-nil here.